### ISSUE TITLE
ItemAccessibilityException not converted into 40X exception

### ISSUE THREAD

**Issue by mvdbeek** [galaxyproject/galaxy#11462]
Came up in https://github.com/galaxyproject/planemo/issues/1143
```
galaxy.webapps.galaxy.api.datasets ERROR 2021-02-24 13:04:42,977 [p:41176,w:3,m:0] [uWSGIWorker3Core0] Error getting display data for dataset (e6693f4d1a4d9f91) from history (e9859c1e537d1ced)
Traceback (most recent call last):
  File "lib/galaxy/webapps/galaxy/api/datasets.py", line 441, in display
    hda = self.hda_manager.get_accessible(decoded_content_id, trans.user)
  File "lib/galaxy/managers/secured.py", line 33, in get_accessible
    return self.error_unless_accessible(item, user, **kwargs)
  File "lib/galaxy/managers/secured.py", line 43, in error_unless_accessible
    raise exceptions.ItemAccessibilityException("%s is not accessible by user" % (self.model_class.__name__))
galaxy.exceptions.ItemAccessibilityException: HistoryDatasetAssociation is not accessible by user
10.10.58.115 - - [24/Feb/2021:13:04:42 +0200] "GET /api/histories/e9859c1e537d1ced/contents/e6693f4d1a4d9f91/display HTTP/1.1" 500 - "-" "python-requests/2.25.1"
```

**Comment by jmchilton** [comment]
Yeah - it still has a ``legacy_expose_api_raw_anonymous`` decorator. Swapping that for the modern one would fix that problem. Probably should fix that exception handling in that method at the same time. I'll open a PR and we will see if anything fails.

**Comment by jmchilton** [comment]
This is really affecting Planemo usability - I'm attaching a paper-cut tag 😉.
